### PR TITLE
Check the validity of the submitted URL more accurately

### PR DESCRIPTION
**Pull request by davidbgk**

### CODE REVIEW

File: croquemort/crawler.py [hunk 1]
```diff
@@ -7,6 +9,12 @@
 from .logger import LoggingDependency
 from .storages import RedisStorage
 
+# See https://github.com/kvesteri/validators for reference.
+url_pattern = re.compile(
+    r'^[a-z]+://([^/:]+\.[a-z]{2,10}|([0-9]{{1,3}}\.)'
+    r'{{3}}[0-9]{{1,3}})(:[0-9]+)?(\/.*)?$'
```
Comment (noirbizarre) [inline croquemort/crawler.py]: What a regexp !!!

**Comment by noirbizarre** [comment]
r+

**Comment by noirbizarre** [comment]
Comment: do we need to blacklist some schemes (smb, gopher... whatever is not relevant) ?

**Comment by davidbgk** [comment]
> do we need to blacklist some schemes

The [Django regexp](http://stackoverflow.com/a/7160778) is more precise about that, I don't think that's a problem for now given that we didn't encounter that use-case. And who knows, that might be relevant for very old systems that are still using ftp ;)

**Pull request merged**